server/src/main/java/org/eclipse/openvsx/AdminService.java

```diff
@@ -233,6 +245,56 @@ public void logAdminAction(UserData admin, ResultJson result) {
             entityManager.persist(log);
         }
     }
+    
+    @Transactional
+    public UserPublishInfoJson getUserPublishInfo(String provider, String loginName) {
+        var user = repositories.findUserByLoginName(provider, loginName);
+        if (user == null) {
+            throw new ErrorResultException("User not found: " + loginName, HttpStatus.NOT_FOUND);
+        }
+        var accessTokens = repositories.findAccessTokens(user);
+        List<ExtensionJson> versionJsons = new ArrayList<>();
+        List<AccessTokenJson> accessTokenJsons = new ArrayList<>();
+        if (!accessTokens.isEmpty()) {
+            for (var accessToken : accessTokens) {
+                var versions = repositories.findVersionsByAccessToken(accessToken);
+                accessTokenJsons.add(accessToken.toAccessTokenJson());
+                for (var version : versions) {
+                    versionJsons.add(local.toExtensionVersionJson(version));
+                }
+            }
+        }
+        var userPublishInfo = new UserPublishInfoJson();
+        userPublishInfo.user = user.toUserJson();
+        userPublishInfo.extensions = versionJsons;
+        userPublishInfo.accessTokens = accessTokenJsons;
+        return userPublishInfo;
+    }
+
+    @Transactional(rollbackOn = ErrorResultException.class)
+    public ResultJson revokePublisherAgreement(String provider, String loginName, UserData admin) {
+        var user = repositories.findUserByLoginName(provider, loginName);
+        if (user == null) {
+            throw new ErrorResultException("User not found: " + loginName, HttpStatus.NOT_FOUND);
+        }
+        int numberOfTokens = changePublishersAccessTokensActivity(user, false);
+        var userPublishInfo = getUserPublishInfo(provider, loginName);
+        for (var extension : userPublishInfo.extensions) {
+            deleteExtension(extension.namespace, extension.name, extension.version, admin);
+        }
+        return ResultJson.success("Set activity of " + numberOfTokens + " tokens and deleted " + userPublishInfo.extensions.size() + " extensions."); 
```

Comment:
Please also add a log entry via `logAdminAction`. The entry message should contain the provider/loginName of the affected user.